avaframe/ana5Utils/distanceTimeAnalysis.py

```diff
@@ -0,0 +1,653 @@
+        message = 'Radar location outside of DEM, x= %.2f, %.2f, y= %.2f, %.2f' % (radarFov[0][0],
+            radarFov[0][1], radarFov[1][0], radarFov[1][1])
+        log.error(message)
+        raise AssertionError(message)
+
+    # define radar field of view as dict
+    radarpath = {'x': np.asarray(radarFov[0]),
+                 'y': np.asarray(radarFov[1])}
+    # add z-coordinate from dem data
+    radarProfile = gT.projectOnRaster(demOriginal, radarpath, interp='bilinear')
+
+    # radar location - position of radar [0] and point in direction of field of view [1]
+    rX = radarProfile[0]['x'][0]
+    rY = radarProfile[0]['y'][0]
+    rZ = radarProfile[0]['z'][0]
+    # a point that gives direction - point in direction of field of view
+    rDx = radarProfile[0]['x'][1]
+    rDy = radarProfile[0]['y'][1]
+    rDz = radarProfile[0]['z'][1]
+
+    # convert centerline of radar field of view to spherical coordinates
+    rD, phiD, thetaD = gT.cartToSpherical(X=rDx-rX, Y=rDy-rY, Z=rDz-rZ)
+
+    # domain in spherical coordinates with center beeing radar position
+    r, phi, theta = gT.cartToSpherical(X=X-rX, Y=Y-rY, Z=rasterdata-rZ)
+
+    # create mask where in spherical coordinates azimuth angle is bigger than field of view of radar
+    # this is defined by the radar position where it is looking and the aperature angle
+    maskPhi = ~np.logical_and(phi > phiD - aperture, phi < phiD + aperture)
+
+    # compute radar range field (distance from radar -masked with field of view)
+    # TODO: check if we need to mask also where there are nans in dem
+    radarRange = np.ma.array(r, mask=maskPhi)
+
+    # create range gates
+    minR = int(np.floor(np.nanmin(radarRange)))
+    maxR = int(np.ceil(np.nanmax(radarRange)))
+    rgWidth = cfgRangeTime['GENERAL'].getfloat('rgWidth')
+    rangeGates= np.arange(minR+rgWidth/2, maxR+rgWidth/2, rgWidth)
+
+    # create plot of range distance already masked with radar field of view
+    dtAnaPlots.radarFieldOfViewPlot(radarFov, aperture, radarRange, X, Y, cfgRangeTime['GENERAL'],
+        rangeGates, demOriginal)
+
+    return radarRange, rangeGates
+
+
+def minRangeSimulation(flowF, rangeMasked, threshold):
+    """ fetch info on the min range of simulation with respect to masked range array
+        the masked range array gives info what the visible range is
+
+        Parameters
+        -------------
+        flowF: numpy nd array
+            flow parameter result field
+        rangeMasked: masked numpy array
+            masked array of radar range
+        threshold: float
+            used to create mask for getting line of sight range,
+            data below threshold not taken into account
+
+        Returns
+        --------
+        losRange: float
+            minimum distance in line of sight to avalanche front
+            defined by the flowF
+
+    """
+
+    # fetch mask where result field smaller than threshold
+    maskAva = np.ma.masked_where(flowF < threshold, flowF)
```

Comment:
I guess if you use 
maskAva = np.ma.masked_where(flowF <= threshold, flowF)
and have a threshold = 0
You do not have to trick with threshold = 0.00001
This would apply everywhere

Reply:
I depends a bit on the stability of the result field. Are we sure, that outside an avalanche the result field is always numerically 0? Or do sometime very small numbers exist? We could also do "maskAva = np.ma.masked_where(flowF <= threshold + epsilon, flowF) with threshold = 0 and epsilon = sys.float_info.epsilon" to prevent problems with float comparison (https://stackoverflow.com/a/9528651)

Reply:
Question about the threshold applied:
1. Is this always related to the machine precision? 
2. Or is it related to also include the impreciseness that comes from the model itself? Like eg for certain questions/tasks one ignores eg. everything where release thickness is below 10cm or something because for the question at hand those values are irrelevant.?

We are talking about relative vs absolute error  (good comparison here: https://codingnest.com/the-little-things-comparing-floating-point-numbers/#fn13). And while @snowtechblog  proposed solution is right from a technical / purely mathematical standpoint (under the assumption that all our results are precise down to machine precision), in reality this is irrelevant. Another good resource is https://floating-point-gui.de/errors/comparison/ 
There is a reason why functions like https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html have both rtol and atol as parameters. 

Or to rephrase it: is it even remotely relevant to look at something to the e-16 for closeness when we know that we introduce, as @snowtechblog said, those 'very small numbers' because of some numerical issue? like for example the 'unstoppable particles' due to friction (among many others)?

So what I propose is to use a tolerance threshold value that makes sense in regards to the data we are looking at. And maybe to make this more obvious the first proposal by @matthiasto about masking by looking for values ABOVE the threshold makes the most sense...